### PR TITLE
Comment ArmsItem

### PR DESCRIPTION
**Pull request by GabrielRavier**
Just some general documentation of the code

### CODE REVIEW

File: src/ArmsItem.cpp [hunk 1]
```diff
@@ -158,8 +172,10 @@ BOOL SubItemData(long code)
 	return TRUE;
 }
 
-void MoveCampCursor()
+/// Update the inventory cursor
+static void MoveCampCursor()
```
Comment (Clownacy) [inline src/ArmsItem.cpp]: If I remember right, marking functions as static causes them to be placed at the start of the object file, breaking the function ordering in the EXE.

File: src/ArmsItem.cpp [hunk 2]
```diff
@@ -313,17 +355,19 @@ void PutCampObject()
...
 
-		rcArms.left = 16 * (gArmsData[i].code % 16);
-		rcArms.right = rcArms.left + 16;
-		rcArms.top = 16 * (gArmsData[i].code / 16);
-		rcArms.bottom = rcArms.top + 16;
+		// Get icon rect for next weapon
+		rcArms.left = TILES_TO_PIXELS(gArmsData[i].code % 0x10);
+		rcArms.right = rcArms.left + TILES_TO_PIXELS(1);
+		rcArms.top = TILES_TO_PIXELS(PIXELS_TO_TILES(gArmsData[i].code));
+		rcArms.bottom = rcArms.top + TILES_TO_PIXELS(1);
```
Comment (Clownacy) [inline src/ArmsItem.cpp]: I don't think this is right. These calculations aren't related to tiles.

File: src/ArmsItem.cpp [hunk 3]
```diff
@@ -337,22 +381,26 @@ void PutCampObject()
...
 
 	// Draw items cursor
 	if (gCampActive == TRUE)
-		PutBitmap3(&rcView, 32 * (gSelectedItem % 6) + (WINDOW_WIDTH - 224) / 2, 16 * (gSelectedItem / 6) + (WINDOW_HEIGHT - 88) / 2, &rcCur2[(flash / 2) % 2], SURFACE_ID_TEXT_BOX);
+		PutBitmap3(&rcView, 32 * (gSelectedItem % 6) + (WINDOW_WIDTH - 224) / 2, 16 * (gSelectedItem / 6) + (WINDOW_HEIGHT - 88) / 2, &rcCur2[(flash / 2) % 2],
+		           SURFACE_ID_TEXT_BOX);
 	else
-		PutBitmap3(&rcView, 32 * (gSelectedItem % 6) + (WINDOW_WIDTH - 224) / 2, 16 * (gSelectedItem / 6) + (WINDOW_HEIGHT - 88) / 2, &rcCur2[1], SURFACE_ID_TEXT_BOX);
+		PutBitmap3(&rcView, 32 * (gSelectedItem % 6) + (WINDOW_WIDTH - 224) / 2, 16 * (gSelectedItem / 6) + (WINDOW_HEIGHT - 88) / 2, &rcCur2[1],
+		           SURFACE_ID_TEXT_BOX);
```
Comment (Clownacy) [inline src/ArmsItem.cpp]: This isn't how I prefer to style these things.

File: src/ArmsItem.cpp [hunk 4]
```diff
@@ -392,36 +443,39 @@ int CampLoop()
...
 			switch (Call_Escape(ghWnd))
 			{
-				case 0:
-					return 0;
-				case 2:
-					return 2;
+			case 0:
+				return 0; // Quit game
+			case 2:
+				return 2; // Go to game intro
```
Comment (Clownacy) [inline src/ArmsItem.cpp]: Also not my style.

File: src/ArmsItem.cpp (hunk 1 above)
Comment (GabrielRavier) [reply src/ArmsItem.cpp]: Fixed

File: src/ArmsItem.cpp (hunk 3 above)
Comment (GabrielRavier) [reply src/ArmsItem.cpp]: Was just trying to avoid very long lines, what should I do about them ?

File: src/ArmsItem.cpp (hunk 4 above)
Comment (GabrielRavier) [reply src/ArmsItem.cpp]: You mean you indent before a switch case ?

Comment (GabrielRavier) [reply src/ArmsItem.cpp]: Pushed a commit that adds indentation before the switch cases

File: src/ArmsItem.cpp (hunk 3 above)
Comment (Clownacy) [reply src/ArmsItem.cpp]: I just leave them long.

File: src/ArmsItem.cpp [hunk 5]
```diff
@@ -70,17 +79,18 @@ BOOL AddArmsData(long code, long max_num)
...
 	for (i = 0; i < ARMS_MAX; ++i)
 		if (gArmsData[i].code == code)
-			break;
+			break; // Found
 
 #ifdef FIX_BUGS
 	if (i == ARMS_MAX)
 #else
-	if (i == ITEM_MAX)	// Oops
+	if (i == ITEM_MAX) // Wrong
```
Comment (Clownacy) [inline src/ArmsItem.cpp]: This is nitpicky, but I only indent comments with spaces when I want to align them with each other. For one-off comments, I indent with a single tab.

File: src/ArmsItem.cpp [hunk 6]
```diff
@@ -271,19 +297,35 @@ void MoveCampCursor()
 	}
 }
 
-void PutCampObject()
+/// Draw the inventory
+static void PutCampObject()
```
Comment (Clownacy) [inline src/ArmsItem.cpp]: Another function that was set as static.

File: src/ArmsItem.cpp (hunk 3 above)
Comment (GabrielRavier) [reply src/ArmsItem.cpp]: Reversed line-shortening

File: src/ArmsItem.cpp (hunk 6 above)
Comment (GabrielRavier) [reply src/ArmsItem.cpp]: Fixed

File: src/ArmsItem.cpp (hunk 5 above)
Comment (GabrielRavier) [reply src/ArmsItem.cpp]: Fixed

**Comment by Clownacy** [comment]
Looks great, thanks. I'll just have to check it's ASM-accurate before I merge it.

**Pull request merged**